### PR TITLE
Fence: move copter fence_check to AP_Arming (1/3)

### PR DESCRIPTION
**Pull request by khancyr**
and prepare to be used by rover

### CODE REVIEW

**Comment by khancyr** [comment]
rebased and correct ordering warning

**Comment by khancyr** [comment]
rebased and correct error on plane

**Comment by khancyr** [comment]
rebased on master

**Comment by rmackay9** [comment]
i'm not really a fan of adding a dependency on AC_Fence from AP_Arming...

File: libraries/AP_Arming/AP_Arming.cpp [hunk 1]
```diff
@@ -466,6 +467,26 @@ bool AP_Arming::board_voltage_checks(bool report)
     return true;
 }
 
+bool AP_Arming::fence_checks(bool report)
+{
+    if ((checks_to_perform & ARMING_CHECK_ALL) ||
+        (checks_to_perform & ARMING_CHECK_FENCE)) {
```
Comment (muramura) [inline libraries/AP_Arming/AP_Arming.cpp]: This processing is done when the ALL bit or FENCE bit is set in the setting value. I think that it is better to code this as a code.

**Comment by khancyr** [comment]
Close as we won't add fence to AP_Arming now

**Pull request closed without merging**